biodatasets/epicqa/epicqa.py

```diff
@@ -0,0 +1,295 @@
+                    "contexts": [
+                        {
+                            "section": datasets.Value("string"),
+                            "context_id": datasets.Value("string"),
+                            "text": datasets.Value("string"),
+                            "sentences": [
+                                {
+                                    "sentence_id": datasets.Value("string"),
+                                    "start": datasets.Value("int32"),
+                                    "end": datasets.Value("int32"),
+                                },
+                            ],
+                        }
+                    ],
+                }
+            )
+        
+        elif self.config.name == "epicqa_research_articles_source":
+
+            features = datasets.Features(
+                {
+                    "document_id": datasets.Value("string"),
+                    "metadata": {
+                        "title": datasets.Value("string"),
+                        "authors": datasets.Value("string"),
+                        "urls": datasets.Sequence(datasets.Value("string")),
+                        "full_text_path": datasets.Value("string"),
+                    },
+                    "contexts": [
+                        {
+                            "section": datasets.Value("string"),
+                            "context_id": datasets.Value("string"),
+                            "text": datasets.Value("string"),
+                            "sentences": [
+                                {
+                                    "sentence_id": datasets.Value("string"),
+                                    "start": datasets.Value("int32"),
+                                    "end": datasets.Value("int32"),
+                                },
+                            ],
+                        }
+                    ],
+                }
+            )
+
+        # Simplified schema for QA tasks
+        else:
+
+            features = schemas.qa_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+
+        my_urls = _URLs[self.config.name]
+        data_dir = dl_manager.download_and_extract(my_urls)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": data_dir,
+                    "split": "train",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split):
+        """Yields examples as (key, example) tuples."""
+
+        subset_name = "research_articles" if "research_articles" in self.config.name else "consumer_articles"
```

Comment:
can we make this a little more explicit? 
```suggestion
        if self.config.subset_id.endswith("research_articles"):
            subset_name = "research_articles"
        elif self.config.subset_id.endswith("consumer_articles"):
            subset_name = "consumer_articles"
        else:
            raise ValueError(f"subset_id {self.config.subset_id} does not end with 'research_articles' or 'consumer_articles')
```